### PR TITLE
[FlashRuntimeExtensions] Add new port

### PR DESCRIPTION
**Pull request by al-sabr**
**Describe the pull request**

- #### What does your PR fix?  
Windows port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
Windows x86 & x64

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

Submission containing materials of a third party: Apache Flex Framework

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@al-sabr ,Thanks for your pr, could you please sign CLA first?

**Comment by al-sabr** [comment]
> @al-sabr ,Thanks for your pr, could you please sign CLA first?

Wher ?

**Comment by JonLiu1993** [comment]
> > @al-sabr ,Thanks for your pr, could you please sign CLA first?
> 
> Where ?

There are hints above

**Comment by al-sabr** [comment]
I've read and accepted! Thank you for feedback.

File: ports/flash-runtime-extensions/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO al-sabr/FlashRuntimeExtensions
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/portfile.cmake:5]: @al-sabr, Thanks for your pr, why are we using a private repo, I can’t find this repo on github

File: ports/flash-runtime-extensions/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO al-sabr/FlashRuntimeExtensions
+    REF v2.3
+    SHA512 461e6bb7ec49cb7d687d6504f9b6261eca407c2a79ba29b59abfaf48625e4e077640b43b933fdefe3bdaa76a283cbf2c4495220534c41ca982b7c370ea73f9e5
+    HEAD_REF main
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/portfile.cmake:13]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/flash-runtime-extensions/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+vcpkg_install_cmake()
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/portfile.cmake:16]:
```suggestion
vcpkg_cmake_install()
```

File: ports/flash-runtime-extensions/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/portfile.cmake:18]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

File: ports/flash-runtime-extensions/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "flash-runtime-extensions",
+  "version": "2.3",
+  "port-version": 0,
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/vcpkg.json:4]:
```suggestion
```

File: ports/flash-runtime-extensions/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "flash-runtime-extensions",
+  "version": "2.3",
+  "port-version": 0,
+  "description": "Adobe AIR Native Extension API for Windows/MacOS/Linux",
+  "homepage": "https://github.com/facebookresearch/flashlight",
+  "supports": "!(linux | osx)",
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/vcpkg.json:7]:
```suggestion
  "supports": "!(linux | osx)",
  "dependencies": [
   {
     "name": "vcpkg-cmake",
     "host": true
   },
   {
     "name": "vcpkg-cmake-config",
     "host": true
   }
  ]

```

File: ports/flash-runtime-extensions/README.md [hunk 7]
```diff
@@ -0,0 +1,2 @@
+# flash-runtime-extensions
```
Comment (JonLiu1993) [inline ports/flash-runtime-extensions/README.md:1]: We do not recommend adding README.md to vcpkg，please remove it

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: versions/f-/flash-runtime-extensions.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "a9ee479db50af850b836b328ba6879088a47a22c",
+      "version-string": "2021-12-23",
```
Comment (JonLiu1993) [inline versions/f-/flash-runtime-extensions.json:5]: Why is version-string here, we should be consistent with the version type in vcpkg.json

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/flash-runtime-extensions/vcpkg.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "flash-runtime-extensions",
+  "version": "2.3",
+  "port-version": 0,
+  "description": "Adobe AIR Native Extension API for Windows/MacOS/Linux",
+  "homepage": "https://github.com/facebookresearch/flashlight",
```
Comment (maksqwe) [inline ports/flash-runtime-extensions/vcpkg.json:6]: Incorrect homepage

Comment (al-sabr) [reply ports/flash-runtime-extensions/vcpkg.json]: See https://github.com/al-sabr/vcpkg/commit/84b753917345327829695b91ffd6e348adeef754

Comment (JonLiu1993) [reply ports/flash-runtime-extensions/vcpkg.json]: @al-sabr, It looks like this homepage is not correct, it should be this:
https://github.com/al-sabr/FlashRuntimeExtensions

**Comment by JonLiu1993** [comment]
@al-sabr ,Cou you confirm the reason why CI failed in arm64-windows and arm64-uwp in upstream?

**Comment by al-sabr** [comment]
İ really don't know since I have no arm setup and I cannot see the error logs.

**Comment by maksqwe** [comment]
So do you have source? Or you have headers and .lib file for win32/win64 only?

**Comment by PhoebeHui** [comment]
@al-sabr, it looks the post build step failed, See the CI failures:
```
Starting package 3/3: flash-runtime-extensions:arm64-windows
Building package flash-runtime-extensions[core]:arm64-windows...
-- Downloading https://github.com/al-sabr/FlashRuntimeExtensions/archive/v2.3.tar.gz -> al-sabr-FlashRuntimeExtensions-v2.3.tar.gz...
-- Extracting source D:/downloads/al-sabr-FlashRuntimeExtensions-v2.3.tar.gz
-- Using source at D:/buildtrees/flash-runtime-extensions/src/v2.3-a47b7db0b9.clean
-- Configuring arm64-windows
-- Building arm64-windows-dbg
-- Building arm64-windows-rel
-- Installing: D:/packages/flash-runtime-extensions_arm64-windows/share/flash-runtime-extensions/copyright
-- Performing post-build validation
The following files were built for an incorrect architecture:

    D:\packages\flash-runtime-extensions_arm64-windows\debug\lib\FlashRuntimeExtensions.lib
Expected arm64, but was: x64

    D:\packages\flash-runtime-extensions_arm64-windows\lib\FlashRuntimeExtensions.lib
Expected arm64, but was: x64

Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile:
    C:\a\1\s\ports\flash-runtime-extensions\portfile.cmake
-- Performing post-build validation done
Error: Building package flash-runtime-extensions:arm64-windows failed with: POST_BUILD_CHECKS_FAILED
Elapsed time for package flash-runtime-extensions:arm64-windows: 2.364 s

Triplet: arm64-windows
Total elapsed time: 17.69 s
```

File: ports/flash-runtime-extensions/README.md [hunk 10]
```diff
@@ -0,0 +1,2 @@
+# flash-runtime-extensions
+CMake port of FlashRuntimeExtensions for vcpkg
```
Comment (PhoebeHui) [inline ports/flash-runtime-extensions/README.md:2]: Does this require?

**Comment by PhoebeHui** [comment]
BTW, you can check the logs from CI pipeline:

Install steps:
https://dev.azure.com/vcpkg/public/_build/results?buildId=64933&view=logs&j=84da4f4a-968a-5b5c-16e9-f444c845396d&t=80e5e0c8-bb3b-5216-9f3f-2618d4e9f3f4
image


Download build logs from https://dev.azure.com/vcpkg/public/_build/results?buildId=64933&view=artifacts&pathAsName=false&type=publishedArtifacts


image

**Comment by al-sabr** [comment]
> So do you have source? Or you have headers and .lib files for win32/win64 only?

This is the official link https://github.com/al-sabr/FlashRuntimeExtensions

**Comment by al-sabr** [comment]
At the moment we need to add in this conversation the team behind the development of the SDK. I will create a link to for them to this thread so that they can also share thoughts.

**Comment by al-sabr** [comment]
Is there a way in the json files to specify that arm32 and arm6 is not supported so that it is bypassed for the moment?

File: ports/flash-runtime-extensions/README.md (hunk 10 above)
Comment (al-sabr) [reply ports/flash-runtime-extensions/README.md:2]: Yes! I will put more details in the Future to explain but right now it is rudimentary.

Comment (al-sabr) [reply ports/flash-runtime-extensions/README.md:2]: Yes it does ... I will improve it in Future with more details.

**Comment by JonLiu1993** [comment]
@al-sabr , it looks the post build step failed, 
See the CI failures:
x64-linux:
````
-- Installing: /mnt/vcpkg-ci/packages/flash-runtime-extensions_x64-linux/share/flash-runtime-extensions/copyright
-- Performing post-build validation
There should be no empty directories in /mnt/vcpkg-ci/packages/flash-runtime-extensions_x64-linux
The following empty directories were found:

    /mnt/vcpkg-ci/packages/flash-runtime-extensions_x64-linux/debug

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")

Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile:
    /agent/_work/1/s/ports/flash-runtime-extensions/portfile.cmake
-- Performing post-build validation done
Error: Building package flash-runtime-extensions:x64-linux failed with: POST_BUILD_CHECKS_FAILED
Elapsed time for package flash-runtime-extensions:x64-linux: 1.27 s
````

x64-osx:
````
-- Installing: /Users/vagrant/Data/packages/flash-runtime-extensions_x64-osx/share/flash-runtime-extensions/copyright
-- Performing post-build validation
There should be no empty directories in /Users/vagrant/Data/packages/flash-runtime-extensions_x64-osx
The following empty directories were found:

    /Users/vagrant/Data/packages/flash-runtime-extensions_x64-osx/debug

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")

Found 1 post-build check problem(s). To submit these ports to curated catalogs, please first correct the portfile:
    /Users/vagrant/Data/work/1/s/ports/flash-runtime-extensions/portfile.cmake
-- Performing post-build validation done
Error: Building package flash-runtime-extensions:x64-osx failed with: POST_BUILD_CHECKS_FAILED
Elapsed time for package flash-runtime-extensions:x64-osx: 2.266 s
````

**Comment by al-sabr** [comment]
Applied the change.

**Comment by JonLiu1993** [comment]
@al-sabr ,it looks the post build step failed, you can check the logs from CI pipeline:

Install steps:
https://dev.azure.com/vcpkg/public/_build/results?buildId=64933&view=logs&j=84da4f4a-968a-5b5c-16e9-f444c845396d&t=80e5e0c8-bb3b-5216-9f3f-2618d4e9f3f4

Capture3

**Comment by al-sabr** [comment]
> @al-sabr ,it looks the post build step failed, you can check the logs from CI pipeline:
> 
> Install steps:
> https://dev.azure.com/vcpkg/public/_build/results?buildId=64933&view=logs&j=84da4f4a-968a-5b5c-16e9-f444c845396d&t=80e5e0c8-bb3b-5216-9f3f-2618d4e9f3f4
> 
> Capture3
> 

Unfortunately Adobe doesn't provide any debug versions of the libraries. See thread:

**Comment by al-sabr** [comment]
I cannot provide the debug versions of the libraries. They are not available in the SDK.

**Comment by al-sabr** [comment]
> So do you have source? Or you have headers and .lib files for win32/win64 only?

Yes only for win32/win64 nothing else.

**Comment by JonLiu1993** [comment]
> I cannot provide the debug versions of the libraries. They are not available in the SDK.

@al-sabr , Thanks for your diligent investigation, I will discuss this issue with my colleague

File: ports/flash-runtime-extensions/portfile.cmake [hunk 11]
```diff
@@ -0,0 +1,22 @@
+#set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO al-sabr/FlashRuntimeExtensions
```
Comment (JackBoosY) [inline ports/flash-runtime-extensions/portfile.cmake:5]: Does this extension only provide binary? Can we use source code to compile? Is it provided by adobe?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/flash-runtime-extensions/portfile.cmake (hunk 11 above)
Comment (al-sabr) [reply ports/flash-runtime-extensions/portfile.cmake:5]: The headers are open but the binaries are closed source. See previous threads for clarifications.

Comment (al-sabr) [reply ports/flash-runtime-extensions/portfile.cmake:5]: See https://github.com/airsdk/Adobe-Runtime-Support/issues/1490

File: ports/flash-runtime-extensions/portfile.cmake [hunk 12]
```diff
@@ -0,0 +1,22 @@
+#set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
```
Comment (JackBoosY) [inline ports/flash-runtime-extensions/portfile.cmake:1]:
```suggestion
set(VCPKG_POLICY_ONLY_RELEASE_CRT enabled)
```

File: ports/flash-runtime-extensions/portfile.cmake [hunk 13]
```diff
@@ -0,0 +1,22 @@
+set(VCPKG_POLICY_ONLY_RELEASE_CRT enabled)
```
Comment (JackBoosY) [inline ports/flash-runtime-extensions/portfile.cmake:1]:
```suggestion
set(VCPKG_POLICY_MISMATCHED_NUMBER_OF_BINARIES enabled)
```

File: ports/flash-runtime-extensions/vcpkg.json [hunk 14]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "flash-runtime-extensions",
+  "version": "2.4",
+  "description": "Adobe AIR Native Extension API for Windows/MacOS/Linux",
+  "homepage": "https://github.com/al-sabr/FlashRuntimeExtensions",
+  "supports": "!(arm64 | (arm & uwp))",
```
Comment (JackBoosY) [inline ports/flash-runtime-extensions/vcpkg.json:6]:
```suggestion
  "supports": "!(arm | uwp)",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/flash-runtime-extensions/vcpkg.json (hunk 14 above)
Comment (JackBoosY) [reply ports/flash-runtime-extensions/vcpkg.json:6]: All the arm or uwp triplets are not supported, right?

Comment (al-sabr) [reply ports/flash-runtime-extensions/vcpkg.json:6]: Right

**Comment by JonLiu1993** [comment]
@al-sabr ，Please run command "./vcpkg x-add-version flash-runtime-extensions --overwrite-version" and commit again

**Comment by al-sabr** [comment]
> vcpkg x-add-version flash-runtime-extensions --overwrite-version

Done and pushed !

File: ports/flash-runtime-extensions/vcpkg.json (hunk 14 above)
Comment (JackBoosY) [reply ports/flash-runtime-extensions/vcpkg.json:6]: @al-sabr So please consider my suggestion.

**Comment by al-sabr** [comment]
İs this finally going to be accepted?

**Comment by JackBoosY** [comment]
> İs this finally going to be accepted?

See [my comment](https://github.com/microsoft/vcpkg/pull/22171#discussion_r781817529).

**Comment by al-sabr** [comment]
> > İs this finally going to be accepted?
> 
> See [my comment](https://github.com/microsoft/vcpkg/pull/22171#discussion_r781817529).

Resolved from my side.

File: ports/flash-runtime-extensions/vcpkg.json (hunk 14 above)
Comment (JonLiu1993) [reply ports/flash-runtime-extensions/vcpkg.json:6]: @al-sabr , Please consider the suggestion

**Review by JonLiu1993 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by al-sabr** [comment]
Is this finished?

**Pull request merged**